source/azure_iot_adu_client.c

```diff
@@ -215,15 +209,9 @@ AzureIoTResult_t AzureIoTADUClient_ParseRequest( AzureIoTADUClient_t * pxAzureIo
     {
         if( az_span_size( xBaseUpdateRequest.update_manifest ) > 0 )
         {
-            xAzResult = az_json_string_unescape( xBaseUpdateRequest.update_manifest, ( char * ) az_span_ptr( xBufferSpan ), az_span_size( xBufferSpan ), &lOutManifestSize );
-
-            if( az_result_failed( xAzResult ) )
-            {
-                AZLogError( ( "az_json_string_unescape failed: 0x%08x", xAzResult ) );
-                return AzureIoT_TranslateCoreError( xAzResult );
-            }
+            xBaseUpdateRequest.update_manifest = az_json_string_unescape( xBaseUpdateRequest.update_manifest, xBaseUpdateRequest.update_manifest );
 
-            xAzResult = az_json_reader_init( &jr, az_span_slice( xBufferSpan, 0, lOutManifestSize ), NULL );
+            xAzResult = az_json_reader_init( &jr, xBaseUpdateRequest.update_manifest, NULL );
```

Comment:
Just to confirm:  are we tracking a service feature request to avoid json encoded base64 json documents?